pkg/resolver/rbacHelper.go

```diff
@@ -134,15 +135,23 @@ func matchNamespacedResources(nsResources map[string][]rbac.Resource, userInfo v
 // error if any, while marshaling the resource groups
 func consolidateNsResources(nsResources map[string][]rbac.Resource) (map[string][]string, []string, error) {
 	m := map[string][]string{}
+	mMutex := sync.RWMutex{}
 
 	for ns, resources := range nsResources {
 		b, err := json.Marshal(resources)
 		if err == nil {
-			if _, found := m[string(b)]; found {
+			mMutex.RLock()
+			_, found := m[string(b)]
+			mMutex.RUnlock()
+
+			// Lock map before write
+			mMutex.Lock()
+			if found {
 				m[string(b)] = append(m[string(b)], ns)
 			} else {
 				m[string(b)] = []string{ns}
 			}
+			mMutex.Unlock()
```

Comment:
The lock might cause a problem when reading inside the `if{}`.  This should be safer.
```suggestion
			mMutex.RLock()
			prevValue, found := m[string(b)]
			mMutex.RUnlock()

			// Lock map before write
			mMutex.Lock()
			if found {
				m[string(b)] = append(prevValue, ns)
			} else {
				m[string(b)] = []string{ns}
			}
			mMutex.Unlock()
```

Reply:
I have removed this lock for now. Don't think this is the one causing the error.